cmd/zoekt-dynamic-indexserver/main_test.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+)
+
+var (
+	CmdTimeout = 100 * time.Millisecond
+)
+
+func captureOutput(f func()) string {
+	var buf bytes.Buffer
+	log.SetOutput(&buf)
+	f()
+	log.SetOutput(os.Stderr)
```

Comment:
for mutating global state use defer to ensure you undo it in case of panic (eg log.Fatal uses a panic)
```suggestion
	log.SetOutput(&buf)
	defer func() { log.SetOutput(os.Stderr) }
	f()
```